archive_converter.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import email
+import os
+from setup_gmail import Message
+
+class ArchiveMessageIndex(object):
+    def __init__(self):
+        self._lookup_index = {} # Maps MessageKey to Thread
+        self._messages_seen = {} # Maps message.id to message
+
+    def find_thread(self, key: str) -> Message:
+        return self._lookup_index[key]
+
+    def update(self):
+        fully_updated = False
+        new_messages = []
+
+        for filename in os.listdir('test_data'):
+            if filename.endswith(".txt"):
+                email = generate_email_from_file('test_data/' + filename)
+                if(email.id not in self._messages_seen):
+                    new_messages.append(email)
```

Comment:
nit: reduce nesting
And by that we mean, prefer to return/continue early rather than nesting `if`s deeper and deeper

```suggestion
            if not filename.endswith(".txt"):
                continue
            email = generate_email_from_file('test_data/' + filename)
            if(email.id not in self._messages_seen):
                new_messages.append(email)
```

It's a bit easier to read since we don't not have to keep in mind how deep we are.
Readers can think "oh, I can assume it's a .txt file for the lines after this" instead of "it's a .txt file in this nested block"